### PR TITLE
Fixed page title inconsistencies

### LINKED ISSUE #350: Page title inconsistencies
After #333, most page titles are awesome. Good work @dariusf.

Problem now is there are some pages with custom titles. Look at the course users controller - the page headings differ from the page titles. That isn't reflected in the page title.

Also, it'll be a good idea to test the page title application helper while you're working through this problem. There's an extra line which is not used, and not covered.

### PR DESCRIPTION
**Pull request by dariusf**
Fixes #350. All actions now have breadcrumbs (and thus page titles). `page_title` helper is also covered with tests now.

### CODE REVIEW

**Comment by dariusf** [comment]
Maybe this can depend on #357, then I can do groups as well.

**Comment by lowjoel** [comment]
Can this be DRYed up using a controller before_action?

**Comment by dariusf** [comment]
I don't think so. There are exceptions to what happens normally on both sides of the hierarchy (the component controller on the `System::Admin::Controller` side, and the user-related ones on the `Course::Controller` side).

**Comment by lowjoel** [comment]
For new it definitely can.

For edit you might want to define a humaniser method, e.g. to_s or to_title or to_display or something to that effect.

I do not fancy writing these methods all over the controller code. It's very easy to forget.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -82,7 +88,7 @@ The `lib` directory is not autoloaded, as described in this [blog post](http://h
 directory is. Place libraries which will be autoloaded in that directory instead.
 
 ## Nested layouts
-Following a paradigm similar to that described by [Andr� Arko](http://andre.arko.net/2013/02/02/nested-layouts-on-rails--31/),
+Following a paradigm similar to that described by [Andr� Arko](http://andre.arko.net/2013/02/02/nested-layouts-on-rails--31/),
```
Comment (allenwq) [inline CONTRIBUTING.md]: Find a encoding problem here @dariusf

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -42,11 +42,10 @@ This allows models to be inherited. See the section on _Inherited Callback Queue
...
 ## Views
-Remember to specify page breadcrumbs (using the `add_breadcrumb` helper in controllers. Page titles
-will be generated by reversing breadcrumbs, so the name of current resource will always be clearly
-present at the start of the title (for users to quickly switch to the tab when many tabs are open).
-This will also allow users to jump to a page by the location in the module. The Coursemology
-branding is automatically appended to all page titles.
+Page titles will be generated by reversing breadcrumbs, so the name of current resource will always
+be clearly present at the start of the title (for users to quickly switch to the tab when many tabs
+are open). This will also allow users to jump to a page by the location in the module. The
+Coursemology branding is automatically appended to all page titles.
```
Comment (lowjoel) [inline CONTRIBUTING.md]: Say that programmers can still manually override the page title if necessary, and how to do so.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -60,6 +59,13 @@ When displaying translations for long stretches of text (e.g. a paragraph), use
 
 When using simple_form remember to declare f.error_notification.
 
+## Controllers
+Remember to specify page breadcrumbs in controllers with the `add_breadcrumb` helper.
+
+Controllers which involve resources should have breadcrumbs for all relevant actions. Actions which
+involve resources individually (`show`, `edit`) should have breadcrumb text which describes the
```
Comment (lowjoel) [inline CONTRIBUTING.md]: the Rails term for this is called 'resource members'

File: app/controllers/course/users_controller.rb [hunk 4]
```diff
@@ -5,18 +5,22 @@ class Course::UsersController < Course::ComponentController
   add_breadcrumb :index, :course_users_students_path
 
   def students # :nodoc:
+    add_breadcrumb :students
```
Comment (lowjoel) [inline app/controllers/course/users_controller.rb]: Shouldn't these be in the views then?

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -63,6 +67,13 @@ When displaying translations for long stretches of text (e.g. a paragraph), use
 
 When using Simple Form remember to declare `f.error_notification`.
 
+## Controllers
+Remember to specify page breadcrumbs in controllers with the `add_breadcrumb` helper.
+
+Controllers which involve resources should have breadcrumbs for all relevant actions. Actions which
+involve a resource member (`show`, `edit`) should have breadcrumb text which describes the resource,
+e.g. showing its title or name.
```
Comment (lowjoel) [inline CONTRIBUTING.md]: and this should be declared in the view

**Pull request merged**